src/main/java/io/odpf/kafkaclient/ClientFactory.java

```diff
@@ -1,4 +1,31 @@
 package io.odpf.kafkaclient;
 
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Properties;
+
 public abstract class ClientFactory {
```

Comment:
This is not a good design IMO. Some of the issues that I can see :
1. This is not exactly a Factory. I think decide to use a factory makes sense here. But Factory should give concrete Implementation details. The task of factory should be to decide whether it should be producer or consumer based on some condition. Basically this piece of code in main class 
```
ClientFactory clientFactory = (args[0].equals("producer")) ?
                new ProducerFactory() : new ConsumerFactory();
```
Read more on Factory here. https://www.tutorialspoint.com/design_pattern/factory_pattern.htm
2. This defies the single responsibility principle. First ask your self what is the responsibility of this class and it should be minimal + concrete. Here the responsibility is parsing properties text or creating client.
3. The client could have been made interface instead of abstract class. Avoid abstract classes and go for interfaces. Since interfaces are contracts and seggregates implementation.

Reply:
Abstract class was needed to avoid duplicate code in Producer and Consumer, so that it is easy to maintain and develop in future.